application/src/main/java/org/thingsboard/server/service/security/auth/mfa/config/DefaultTwoFaConfigManager.java

```diff
@@ -179,6 +180,7 @@ public PlatformTwoFaSettings savePlatformTwoFaSettings(TenantId tenantId, Platfo
     }
 
     @Override
+    @Transactional
```

Comment:
Check the class location! It is not a DAO layer. It can not be Transactional. this has to call the DAO adminSettingsService instead adminSettingsDao. And adminSettingsService.removeById have to be transactional

Reply:
adminSettingsDao.removeById
- only called from deletePlatformTwoFaSettings
- deletePlatformTwoFaSettings is not called anywhere

Reply:
I agree with @smatvienko-tb comments. If the method is not used - it may be removed from the code base. But if the method is used, it needs to be implemented according to the best practices and patterns. Actually, we might need to create a method called removeByKey or similar.